### PR TITLE
feat: add `Validate` method to `InitialGenesis`

### PR DESCRIPTION
**Pull request by lubtd**
Add a method for validation in the `InitialGenesis` interface to have a stateless check on the values (non empty genesis URL for example)
Add additional tests for this

### CODE REVIEW

File: testutil/sample/launch.go [hunk 1]
```diff
@@ -136,3 +138,9 @@ func MsgEditChain(
 		initialGenesis,
 	)
 }
+
+// GenesisHash returns a sample sha256 hash of custom genesis for GenesisURL
+func GenesisHash() string {
```
Comment (Pantani) [inline testutil/sample/launch.go:143]: If we merge the #230 first, we can reuse this function:

https://github.com/tendermint/spn/pull/230/files#diff-6619c7d1a9a468c88892c0e3dcd2961584eae593aae45d7071bc902b9eb59896R13-R18

File: x/launch/types/initial_genesis.go [hunk 2]
```diff
@@ -1,4 +1,28 @@
...
+// Validate implements InitialGenesis
+func (g GenesisURL) Validate() error {
+	if g.Url == "" {
+		return errors.New("no url provided")
+	}
+	if len(g.Hash) != HashLength {
+		return errors.New("hash must be sha256")
+	}
+	return nil
+}
```
Comment (Pantani) [inline x/launch/types/initial_genesis.go:28]: Should we create separate tests for it? What do you think?

**Review by Pantani — APPROVED** [review]
🙌🏽
Only a few suggestions

File: x/launch/types/initial_genesis.go (hunk 2 above)
Comment (lubtd) [reply x/launch/types/initial_genesis.go:28]: Yep good point. We should maybe think about having a test coverage CI step

Comment (Pantani) [reply x/launch/types/initial_genesis.go:28]: we can use the [codecov](https://about.codecov.io/language/go/)

**Review by ilgooz — DISMISSED** [review]

**Review by Pantani — APPROVED** [review]
👏🏼

**Pull request merged**